photon-api/src/main/scala/com/linkedin/photon/ml/model/MatrixFactorizationModel.scala

```diff
@@ -54,20 +55,52 @@ class MatrixFactorizationModel(
     }
 
   /**
+   * Check whether two latent factors are the same.
    *
-   * @param dataPoints The dataset to score. Note that the Long in the RDD is a unique identifier for the paired
-   *                   GAMEDatum object, referred to in the GAME code as the "unique id".
-   * @return The score.
+   * @param latentFactors1 The first latent factor
+   * @param latentFactors2 The second latent factor
+   * @return True if the two latent factors are the same, false otherwise
    */
-  override def score(dataPoints: RDD[(Long, GameDatum)]): KeyValueScore = {
-    MatrixFactorizationModel.score(dataPoints, rowEffectType, colEffectType, rowLatentFactors, colLatentFactors)
+  private def isSameLatentFactors(
+      latentFactors1: RDD[(String, Vector[Double])],
+      latentFactors2: RDD[(String, Vector[Double])]): Boolean = {
+
+    latentFactors1.fullOuterJoin(latentFactors2).mapPartitions(iterator =>
+      Iterator.single(iterator.forall { case (_, (factor1, factor2)) =>
+        factor1.isDefined && factor2.isDefined && factor1.get.equals(factor2.get)
+      })
+    ).filter(!_).count() == 0
```

Comment:
I don't think that's possible - do you have a link to the docs of a count function that takes a filter as input?